library/core/src/slice/mod.rs

```diff
@@ -3594,6 +3602,166 @@ impl<T> [T] {
         }
     }
 
+    /// Get a subslice where the first element and the element one past the end is aligned
+    /// to a given alignment.
+    ///
+    /// # Panics
+    ///
+    /// This method requires the alignment to be a multiple (larger than 1) of the alignment of
+    /// the slice element's alignment.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// unsafe {
+    ///     let bytes: [u8; 7] = [1, 2, 3, 4, 5, 6, 7];
+    ///     let ints = bytes.aligned_subslice::<u32>();
+    ///     assert_eq!(ints.len(), 1);
+    /// }
+    /// ```
+    #[must_use]
+    #[unstable(feature = "slice_align_to_ish", issue = "none")]
+    #[inline]
+    pub fn aligned_subslice(&self, align: usize) -> &[T] {
+        let t = crate::mem::size_of::<T>();
+        assert!(
+            t < align,
+            "aligned_subslice does nothing for alignments below or at the element type's alignment"
+        );
+        assert!(
+            align % t == 0,
+            "aligned_subslice only works for alignments that are multiples of the element's size"
+        );
+        let offset = self.as_ptr().addr() % align;
+        // SAFETY: See the `align_to_mut` method for the detailed safety comment.
+        let end_offset = unsafe { self.as_ptr().offset(self.len() as isize) }.addr() % align;
+        let end = self.len() - (align / t - end_offset);
+        &self[offset..end]
+    }
+
+    /// Get a subslice where the first element and the element one past the end is aligned
+    /// to a given alignment.
+    ///
+    /// # Panics
+    ///
+    /// This method requires the alignment to be a multiple (larger than 1) of the alignment of
+    /// the slice element's alignment.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// unsafe {
+    ///     let mut bytes: [u8; 7] = [1, 2, 3, 4, 5, 6, 7];
+    ///     let ints = bytes.aligned_subslice_mut::<u32>();
```

Comment:
This doesn't match the signature? It matches what I kind of expected though. But then the function can't be safe any more. Which seems fine?